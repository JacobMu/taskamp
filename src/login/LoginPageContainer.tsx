import React from "react";
import { Card, FormGroup, Button } from "@blueprintjs/core";

import { LoginPageFormView } from "./LoginPageFormView";
import { useLoginPageContainer } from "./useLoginPageContainer";
import "./LoginPageContainer.scss";

const LoginPageContainer: React.FC = () => {
  const { handleSubmit } = useLoginPageContainer();
  return (
    <div className="loginFormLayout">
      <Card>
        <FormGroup>
          <LoginPageFormView />
          <Button onClick={handleSubmit}>{"login.form.confirmBtn"}</Button>
        </FormGroup>
      </Card>
    </div>
  );
};

export default LoginPageContainer;
