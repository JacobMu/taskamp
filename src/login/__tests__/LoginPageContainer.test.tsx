import React from "react";
import { shallow } from "enzyme";
import LoginPageContainer from "../LoginPageContainer";

jest.mock("../useLoginPageContainer", () => ({
  useLoginPageContainer: jest.fn().mockReturnValue({
    handleSubmit: jest.fn().mockName("handleSubmitMock"),
  }),
}));

describe("<LoginPageContainer />", () => {
  it("renders Login page form", () => {
    const wrapper = shallow(<LoginPageContainer />);

    expect(wrapper).toMatchSnapshot();
  });
});
