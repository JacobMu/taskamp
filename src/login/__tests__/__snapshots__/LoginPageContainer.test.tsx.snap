// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPageContainer /> renders Login page form 1`] = `
<div
  className="loginFormLayout"
>
  <Blueprint3.Card
    elevation={0}
    interactive={false}
  >
    <Blueprint3.FormGroup>
      <LoginPageFormView />
      <Blueprint3.Button
        onClick={[MockFunction handleSubmitMock]}
      >
        login.form.confirmBtn
      </Blueprint3.Button>
    </Blueprint3.FormGroup>
  </Blueprint3.Card>
</div>
`;
